clash-lib/src/Clash/Driver.hs

```diff
@@ -682,55 +686,49 @@ normalizeEntity
   -- ^ Hardcoded 'Type' -> 'HWType' translator
   -> (PrimStep, PrimUnwind)
   -- ^ Hardcoded evaluator (delta-reduction)
-  -> [Id]
+  -> [TopEntityT]
   -- ^ TopEntities
   -> ClashOpts
   -- ^ Debug information level for the normalization process
   -> Supply.Supply
   -- ^ Unique supply
   -> Id
   -- ^ root of the hierarchy
-  -> BindingMap
-normalizeEntity reprs bindingsMap primMap tcm tupTcm typeTrans eval topEntities
+  -> (BindingMap, VarEnv TopEntityT)
```

Comment:
I'm a bit unhappy with out this change turned out. Maybe `TopEntity` should just be part of a `Binding`.

Reply:
But there will only be a single TopEntity in that BindingMap though, since the `normalized` BindingMap only contains the “current” TopEntity, but none of the others (since we don’t renormalize other TopEntities).

Reply:
`sortTop` uses `bindingsMap` to topsort topentities and uses `callGraph :: BIndingMap -> Id -> ..` to figure out the callgraph. So topentities must be in there, no?

Reply:
*all topentities

Reply:
but `runNormalization` returns the `_normalized ` BindingMap found in the `NormalizeState`, not the `_bindings` found in `RewriteState`. So while the latter does contain all topentities, the former does not.

Reply:
Gotcha. It should probably return the whole thing instead. Netlist doesn't seem to rely on _only_ normalized binders in there. As a bonus, it would allow normalization processes to reuse work of previous normalizations. If we insist on keeping this check (so we don't accidentally try to generate netlist for a non-normalized binder) https://github.com/clash-lang/clash-compiler/blob/44af7dfbef8a04224a31f076c837d0df400565d4/clash-lib/src/Clash/Netlist.hs#L203-L205, we could have wrap the term of a `Binding` in:

```haskell
data NTerm = Normalized Term | NotNormalized Term
```